library/core/src/task/wake.rs

```diff
@@ -219,6 +219,13 @@ unsafe impl Sync for Waker {}
 
 impl Waker {
     /// Wake up the task associated with this `Waker`.
+    ///
+    /// Multiple wake-ups (through clones of this `Waker` or `wake_by_ref`) may be
+    /// coalesced into a single `poll` invocation by the runtime, and as long as
+    /// the runtime keeps running and the task is not finished it is expected that
+    /// each wake-up is followed by an invocation of `poll`, even in the absence of
+    /// other events. This makes it possible to yield to other tasks when running
+    /// potentially unbounded processing loops in order to maintain fairness.
```

Comment:
I think this wording to me sounds like what you are getting at is that a call to wake() guarantees at least one call to poll (in the absence of Poll::Ready / runtime termination / panics / etc).

Maybe that's simpler than discussing coalescing etc?

Reply:
My understanding is that if I call `cx.waker().wake_by_ref()` twice in rapid succession, there will usually only be a single wake-up, hence “at least one call” is not correct. But with my proposed “is followed by” wording the coalescing can be left implicit if you find that more appropriate.

Reply:
Yeah, that makes sense. I think I had in mind that the interleaving is such that a wake() ensures at least one future call -- not necessarily *each* wake ensuring a *distinct* future call though.

Reply:
"Guarantees there will be at least one call in the future" sounds correct to me.

Reply:
Thanks for the review, I pushed an improvement that focuses more on the essence.